### PR TITLE
[xcodebuild] Add Spec

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the new behavior (if this is a feature change)?**
Adds `xcodebuild`
**Additional info:**

### CODE REVIEW

**Review by fedeci — COMMENTED** [review]
This command is also available as an `xcrun` subcommand, we should also add it there.

File: dev/xcodebuild.ts [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+        name: "NAME",
+        template: "filepaths",
+      },
+    },
+    {
+      name: "-arch",
+      description:
+        "Build each target for the architecture ARCH; this will override architectures defined in the project",
+      args: {
+        name: "ARCH",
```
Comment (QuiiBz) [inline dev/xcodebuild.ts:83]: Could we provide suggestions for the arch?

File: dev/xcodebuild.ts [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+      name: "-enableThreadSanitizer",
+      description: "turn the thread sanitizer on or off",
+      args: {
+        name: "mode",
+        suggestions: ["YES", "NO"],
+      },
+    },
+    {
+      name: "-enableUndefinedBehaviorSanitizer",
+      description: "turn the undefined behavior sanitizer on or off",
```
Comment (QuiiBz) [inline dev/xcodebuild.ts:246]:
```suggestion
      description: "Turn the address sanitizer on or off",
      args: {
        name: "mode",
        suggestions: ["YES", "NO"],
      },
    },
    {
      name: "-enableThreadSanitizer",
      description: "Turn the thread sanitizer on or off",
      args: {
        name: "mode",
        suggestions: ["YES", "NO"],
      },
    },
    {
      name: "-enableUndefinedBehaviorSanitizer",
      description: "Turn the undefined behavior sanitizer on or off",
```

File: dev/xcodebuild.ts [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+      description:
+        "specifies a path to a plist file that configures archive exporting",
+      args: {
+        name: "PATH",
+        template: "filepaths",
+      },
+    },
+    {
+      name: "-enableCodeCoverage",
+      description: "turn code coverage on or off when testing",
```
Comment (QuiiBz) [inline dev/xcodebuild.ts:324]:
```suggestion
        "Specifies a path to a plist file that configures archive exporting",
      args: {
        name: "PATH",
        template: "filepaths",
      },
    },
    {
      name: "-enableCodeCoverage",
      description: "Turn code coverage on or off when testing",
```

File: dev/xcodebuild.ts [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+    {
+      name: "-skip-testing",
+      description:
+        "Constrains testing by specifying tests to exclude, but including other tests",
+    },
+    {
+      name: "-skip-testing:",
+      insertValue: "-skip-testing:{cursor}",
+      description:
+        "Constrains testing by specifying tests to exclude, but including other tests",
```
Comment (QuiiBz) [inline dev/xcodebuild.ts:418]: Can this option exclude `-only-testing:` and `-only-testing`?

File: dev/xcodebuild.ts [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+    },
+    {
+      name: "-allowProvisioningDeviceRegistration",
+      description:
+        "Allow xcodebuild to register your destination device on the developer portal if necessary. This flag only takes effect if -allowProvisioningUpdates is also passed.",
+    },
+    {
+      name: "-scmProvider",
+      description:
+        "Which implementation to use for Git operations (system/xcode)",
```
Comment (QuiiBz) [inline dev/xcodebuild.ts:494]: I think this option takes an arg

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Mostly uppercase at the beginning of sentences to fix

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**